src/main/java/core/basesyntax/operations/Buy.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.dao.Box;
+import core.basesyntax.dao.Storage;
+import core.basesyntax.goods.Product;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Buy implements Operation {
+    @Override
+    public boolean updateStorage(Product product, Storage storage) {
+        LocalDate productExpDate = product.getExpDate();
+        Product.checkExpDate(productExpDate);
```

Comment:
Don't you want to use try/catch block here? You can get an exception

Reply:
I've got a static check method for throwing an exception. I expect an exception in here in case of improper usage of the method.